tsdb/engine/tsm1/engine.go

```diff
@@ -1193,7 +1195,7 @@ func (e *Engine) overlay(r io.Reader, basePath string, asNew bool) error {
 			return err
 		}
 	}
-	return nil
+	return e.MeasurementFieldSet().WriteToFile()
```

Comment:
Do we call this during ImportShard / anywhere else in an 'online' context? Is this safe?

Reply:
`overlay` obtains a write lock on the engine. It is called from `Import` and `Restore`.  Theoretically, no writes can take place, and this is noted in comments in both call sites.

Reply:
@lesam - Did the explanation above satisfy your concern?

Reply:
Yes - we lock on it now which makes it safe.